### DISCUSSION TITLE
Current hold-to-confirm behaviour doesn't match stable

### DISCUSSION THREAD

**Discussion by LittleEndu** [ppy/osu#15562] [Q&A]
(This issue is now split between #6397, #15334, and #15256 which all have accepted answers that result in them referring the issue back to each other in a cycle. Making this discussion to actually have a place to discuss if and how this behaviour should be implemented.)

Right now the "Hold-to-confirm activation time" setting controls the time of many things. When it comes to gameplay, two major things it controls are going to pause menu and quick retry. The issue comes from the fact that on stable going to pause menu is instant while quick retry is hold-to-confirm. There is no way to achieve this behaviour on lazer.

From previously made discussions threads it is clear that a split between the hold-to-confirm times of the back key and the quick retry key is wanted. Looking at the code, quick retry is implemented separately via `HotkeyRetryOverlay` so making it always require the key to be held would be the simplest implementation wise.

**Comment by bdach** [comment]
i don't know what is the point of this discussion other than causing even more noise. the proposal was already formulated quite clearly and every time it was explained that the new behaviour is intentional and there were *many* people coming from stable who wanted a delay on pausing to avoid pausing accidentally. and i also see sentiments from the mania community that pauses should cause a reduction in pp gained, or even take away all pp entirely, which could exacerbate this concern. so i don't see what else there is to discuss.

the option is there to go back to instant. something "not matching stable" is not an automatic argument that it should be added to lazer.

**Reply by LittleEndu** [reply]
This is missing the point of the three issues mentioned in the OP and there hasn't been any discussion around this. This being that there are user who want pause and retry to have different hold-to-confirm timings. I pointed out it not matching stable because I assume that's what most people are comfortable with: pause being instant and quick retry being delayed. 

I myself rarely pause the game to recover my stamina. I'm just used to every game in existence instantly pausing when you hit <kbd>ESC</kbd>. And for me, <kbd>ESC</kbd> is so far out of the way that accidentally hitting it is out of the question, while my quick retry key is a row higher than my gameplay keys. So putting the global hold-to-confirm setting to one or the other doesn't satisfy me, I either freak out when I go to pause the game but it still keeps playing, or I freak out when I accidentally hit quick retry and instantly lose all progress.

**Comment by peppy** [comment]
There's already many discussions, so let's add ANOTHER one to the pile. How does this make any sense?

Please post your post in one of the existing discussion threads instead.

**Reply by smoogipoo** [reply]
Sounds like one of the threads should be made "unanswered", because at the moment only this one is. And https://github.com/ppy/osu/issues/6397 is locked, which is actually linked to (either directly or indirectly) by the other two discussion threads.

I'm fine with this one existing.
